### ISSUE TITLE
Express Initialization Documentation Out of Date

### ISSUE THREAD

**Issue by Zunawe** [socketio/socket.io#3882]
**Describe the bug**
The [server initialization](https://socket.io/docs/v4/server-initialization/) documentation related to Express still says this is the method to initialize the server is the following:

```js
const io = require("socket.io")(httpServer, options);
```

The correct method is stated in other ways in other places on the page, but since there's a specific Express section it seems like it's a special case, even though the docs are just wrong.

**Comment by darrachequesne** [comment]
Hi! What makes you think it's out of date?

**Comment by Zunawe** [comment]
Hmm, okay, I tried a few more things, checked the source code, and looked through the docs again, here's what caused my confusion:

My project uses TypeScript and Express. When I checked the docs at the above link, I saw a TypeScript section and an Express section. I figured the Express section is more specific to my project, so I'll do what it says there and adapt it to TypeScript with an educated guess. So when I saw

```js
const io = require("socket.io")(httpServer, options);
```

I figured there must be a default export from the module which does something significant to handling Express. But when I used TypeScript to try to get a default export, it couldn't find one. So I fell back to the method described in the TypeScript section and it seemed to work.

Because TypeScript found no default export, I assumed the method I saw in the Express section was outdated and that the socket.io module _used to_ provide a default export for creating Server instances, but it was refactored to be regular export of the module instead because why would the package have a default export that couldn't be detected by TypeScript. The culprit being [this line](https://github.com/socketio/socket.io/blob/master/lib/index.ts#L804), which TypeScript doesn't consider in the type definitions of the export.

All that is probably intentional, which is fine, but when it comes to setting up a project with socket.io for the first time, it's not obvious that the default export is nothing but a Server factory, and that it doesn't exist in the type definition file for TypeScript. Even as you continue to read the docs, it's not explicitly stated that the default export creates an instance of Server, even if it is implied and more obvious as you get more familiar with the library.

---

Given that long-winded explanation of my specific case of confusion, I'll just suggest that it couldn't hurt to explain exactly what is being exported from the `socket.io` module and why TypeScript (and possibly ES modules) won't find a default export. At the moment, without looking at the source code, it's not obvious that the factory function isn't doing anything else behind the scenes.
